### ISSUE TITLE
Allow merging of migration files

### ISSUE THREAD

**Issue by luatnd** [prisma/prisma#21579]
## Problem
I'm using `prisma migrate dev` to generate migration files in my local dev machine and then `prisma migrate deploy` to deploy those migrations to prod.

The problem is you can have so many small & frequent changes to the db schema while you develop the application features.
As a result, migration file1 create a field `foo`, then migration file2 change name `foo` to `bar`, or may be delete `foo`.

That would lead to extremely redundant migration into prod when we run `prisma migrate deploy`
Besides, Prisma has a file validation mechanism that prevents us from modifying the migration files manually.

## Suggested solution

It is better if we can run `prisma migrate dev` only once for each release.

We can cancel the previous migrations, or merge previous migrations into 1 single migration. It's not just concatenating the file, it should provide accumulated migration.

## Alternatives
N/A
Btw, is there any existing supported workflow so I can run `prisma migrate dev` only once for each release?

## Additional context
No

**Comment by jkomyno** [comment]
Hi @luatnd, just to clarify, are asking for a more compact Prisma migration process, that merges all migration files in a single one, and only opens a single transaction to modify the database at once?

**Comment by luatnd** [comment]
> Hi @luatnd, just to clarify, are asking for a more compact Prisma migration process, that merges all migration files in a single one, 

Yes, and migration should remove the redundant, sth like this:

- I have 2 migration files:
  ```sql
  -- add_foo_column/migration.sql
  alter table tb1 add foo integer after user_id;
  
  -- rename_foo_to_bar/migration.sql
  alter table tb1 rename column foo to bar;
  ```
  
  But the final result is actually the same as:
  ```sql
  alter table tb1 add bar integer after user_id;
  ```

- Case 2:
  ```sql
  -- add_foo_column/migration.sql
  alter table tb1 add foo integer after user_id;
  
  -- drop_foo_column/migration.sql
  alter table tb1 drop column foo;
  ```
  The final result should be nothing to do.

- Case 3: Add an index on a large table, then delete them when dev realizes it's not necessary:
  ```sql
  -- add_idx_foo_bar/migration.sql
  create index idx__foo_bar on inv__user_inventory (foo_column desc, bar_column asc);
  
  -- rm_idx_foo_bar_because_of_redundant/migration.sql
  drop index idx__foo_bar;
  ```
  The result should be nothing instead of adding an index and then removing that index.

> and only opens a single transaction to modify the database at once?

No, I don't have any idea about that. In my opinion, multiple tx is better for migration

**Comment by jkomyno** [comment]
Looks like this issue is thus very close to https://github.com/prisma/prisma/issues/5714.

**Comment by Jolg42** [comment]
Closing as duplicate of https://github.com/prisma/prisma/issues/5714
You can comment in that issue if you want to contribute some thoughts.

Maybe you could try out this https://github.com/prisma/prisma/issues/5714#issuecomment-1043163364 and let us know how it goes?
